### PR TITLE
pynodegl: add and use NGL_DLL_DIRS to set additional paths to the DLL search path

### PR DESCRIPTION
**Pull request by mbouron-gpsw**
This will allow the pynodegl module to find more easily the DLLs it
depends on.

### CODE REVIEW

File: pynodegl/pynodegl.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import os
+import platform
+import sysconfig
+
+if platform.system() == 'Windows':
+    ngl_home = os.getenv('NGL_HOME')
+    if ngl_home:
+        os.add_dll_directory(os.path.join(ngl_home, 'venv', 'Scripts')
```
Comment (mbouron-gpsw) [inline pynodegl/pynodegl.py:8]: This needs to be adjusted, venv/Scripts might not useful outside the ngl env itself and will not work on a proper installation.

Comment (mbouron-gpsw) [reply pynodegl/pynodegl.py]: Fixed.

File: pynodegl/pynodegl.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import os
+import platform
+import sysconfig
```
Comment (cboesch-gpsw) [inline pynodegl/pynodegl.py:3]: looks unused

**Comment by cboesch-gpsw** [comment]
Sounds like a good idea; I'm looking forward the integration with the build ecosystem and activate.

File: pynodegl/pynodegl.py (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply pynodegl/pynodegl.py]: what's wrong with my previous solution?

if platform.system() == 'Windows':
    def add_dll_directories():
        dll_dirs = os.getenv("NGL_DLL_DIRS")
        if not dll_dirs:
            return
        dll_dirs = dll_dirs.split(';')
        for p in dll_dirs:
            if os.path.exists(p):
                os.add_dll_directory(p)

Comment (jmoguillansky-gpsw) [reply pynodegl/pynodegl.py]: https://github.com/gopro/gopro-lib-node.gl/pull/272/commits/99cf4d0beee5322daf5929a9150273c088e61c70

Comment (cboesch-gpsw) [reply pynodegl/pynodegl.py]: The issue wasn't in the code (which was reviewed), it was that it was in the wrong place: we needed that for pre-pynodegl, and for all pynodegl users. I believe we explained that it was the issue a few months ago.

Comment (jmoguillansky-gpsw) [reply pynodegl/pynodegl.py]: ok

File: pynodegl/pynodegl.py (hunk 2 above)
Comment (mbouron-gpsw) [reply pynodegl/pynodegl.py:3]: Fixed.

File: pynodegl/pynodegl/__init__.py [hunk 3]
```diff
@@ -19,6 +19,18 @@
...
 #
 
+import os
+import platform
+
+
+if platform.system() == 'Windows':
+    ngl_dll_dirs = os.getenv('NGL_DLL_DIRS')
+    if ngl_dll_dirs:
+        dll_dirs = ngl_dll_dirs.split(';')
```
Comment (jbnoel-gpsw) [inline pynodegl/pynodegl/__init__.py:29]: use `os.pathsep` (https://docs.python.org/3/library/os.html#os.pathsep) instead `';'`

Comment (cboesch-gpsw) [reply pynodegl/pynodegl/__init__.py:29]: Note that this is only for Windows. We also got surprises with Python constants in the past, typically under MSYS2.

Comment (mbouron-gpsw) [reply pynodegl/pynodegl/__init__.py:29]: `os.pathsep` should be fine, it returns `;` on windows native and mingw.

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**